base/libdl.jl

```diff
@@ -205,6 +205,43 @@ if is_linux()
     end
 end # linux-only
 
+if Sys.KERNEL == :FreeBSD
```

Comment:
I checked the manual,  found that they share the same c struct in the first 4 field, but not sure the behaviour is consistent.

I will install VMs and try  later.

NetBSD: http://netbsd.gw.com/cgi-bin/man-cgi?dl_iterate_phdr+3+NetBSD-7.0
OpenBSD: http://man.openbsd.org/OpenBSD-current/man3/dl_iterate_phdr.3
DragonflyBSD: https://www.dragonflybsd.org/cgi/web-man?command=dl_iterate_phdr&section=3